pkg/controllers/state/cluster.go

```diff
@@ -47,9 +48,10 @@ type Cluster struct {
 	clock         clock.Clock
 
 	mu               sync.RWMutex
-	nodes            map[string]*Node                // provider id -> node
-	bindings         map[types.NamespacedName]string // pod namespaced named -> node node
-	nameToProviderID map[string]string               // node name -> provider id
+	nodes            map[string]*Node                 // provider id -> node
+	bindings         map[types.NamespacedName]string  // pod namespaced named -> node node
+	nameToProviderID map[string]string                // node name -> provider id
+	daemonSetPods    map[types.NamespacedName]*v1.Pod // daemonSet -> existing pod
```

Comment:
Can we think about using a `sync.Map` since we don't need to be locking on node stuff while we are accessing this data